### PR TITLE
Convert Implemented Requirements Counts to Dynamic Value Sets

### PR DESCRIPTION
**Pull request by ohsh6o**
As part of #2, in order to properly count `implemented-requirement` and `control` counts, we will want to use XSL functions to pull in implemented requirements.

### CODE REVIEW

**Comment by mike-stern** [comment]
I like the exploration of xslt and how to use its power. The concept of functions makes the complex rules much more readable. I like where it is going.

**Comment by ohsh6o** [comment]
> I like the exploration of xslt and how to use its power. The concept of functions makes the complex rules much more readable. I like where it is going.

I am having a lot of trouble combining it with [abstract patterns](https://www.oxygenxml.com/events/2015/Schematron4ia.dita.pdf) though, which would allow us to soar. We can pair in a bit if you'd like, as it has driven me crazy!

**Review by mike-stern — APPROVED** [review]
After review, i feel good about the reusability and application of the rules. A bit more documentation might be helpful to explain the rules and the reusable parts of the validation, but it's very useful.

File: resources/validations/src/ssp.sch [hunk 1]
```diff
@@ -1,86 +1,235 @@
...
+</xsl:function>
+
+<xsl:function name="lv:sensitivity-level" as="xs:string">
+    <xsl:param name="context" as="node()*"/>
+    <xsl:value-of select="$context//o:security-sensitivity-level"/>
+</xsl:function>
+
+<xsl:function name="lv:profile" as="document-node()*">
+    <xsl:param name="level" />
+    <xsl:variable name="profile-map">
```
[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/src/ssp.sch:91]: As discussed in our pairing for this after formal review, the deal-breaker here is you cannot have `<xsl:key/>` in any context _other than_ the global context of a XSLT stylesheet (or as we have to deal with it, the Schematron top-level that becomes the global context of the resulting XSLT stylesheet. Wrapping these in a function to make testing work and better encapsulate context/state (I mean the term generally for when people look through this stuff again, so I do _not_ mean `<sch:rule context="..."/>` specifically) comes at this cost.

I hope to revisit better encapsulation of global parameters again when xspec/xspec#1294 is merged and xspec/xspec#991 is worked on as well. I am trying to work with upstream devs to know if I can contribute here.

File: resources/validations/bin/validate_with_schematron.sh [hunk 2]
```diff
@@ -79,7 +79,7 @@ fi
 
 #if version not specified default
 SAXON_VERSION=${SAXON_VERSION:-10.2}
-SAXON_OPTS="${SAXON_OPTS:-allow-foreign=true}"
+SAXON_OPTS="${SAXON_OPTS:-allow-foreign=true diagnose=true}"
```
[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/bin/validate_with_schematron.sh:82]: I kept this in here, I think after reading the core Schematron docs this Saxon flag controls whether we will see diagnostics output at build-time or runtime when we do the processing, as source explains [here](https://github.com/Schematron/stf/blob/master/iso-schematron-xslt2/iso_svrl_for_xslt2.xsl#L128-L139) and an example use [here](https://github.com/Schematron/stf/blob/master/iso-schematron-xslt2/iso_svrl_for_xslt2.xsl#L300-L305).

We are not making use of it now, but I think it [will be useful for the future](https://schematron.com/an-overview-of-schematron/why-is-schematron-different/), so I add it in.

**Pull request merged**